pgjdbc/src/main/java/org/postgresql/largeobject/BlobInputStream.java

```diff
@@ -335,6 +351,59 @@ public boolean markSupported() {
     return true;
   }
 
+  /**
+   * Skips over and discards {@code n} bytes of data from this input stream.
+   *
+   * <p>Due to the "sparse" implementation of Large Objects, this class allows skipping
+   * past the "end" of the stream. Subsequent reads will continue to return {@code -1}.
```

Comment:
Could you please explain what do you mean by "sparse" and what do you mean by "skipping past the end of the stream"?

The `skip(...)` contract asks to return the "actual number of bytes skipped". I am leaning towards that if we "skip past the end", we skip no bytes, so the return value should be `0`.

In other words, I would like to know what is the reason for `BlobInputStream#skip` implementation to mismatch `InputStream#skip` when considering "skip past the end".

Reply:
Can you add the link to the comment so that future readers can find it?

Reply:
This is probably a better link: https://www.postgresql.org/docs/14/lo-implementation.html
Unless there are objections regarding the version number?

Reply:
you can always use https://www.postgresql.org/docs/current/lo-implementation.html

Reply:
That tends to break when they move stuff around.